--- a/com/mojang/blaze3d/systems/RenderSystem.java
+++ b/com/mojang/blaze3d/systems/RenderSystem.java
@@ -750,7 +_,7 @@
         Matrix4f matrix4f = new Matrix4f(projectionMatrix);
         if (!isOnRenderThread()) {
             recordRenderCall(() -> {
-                projectionMatrix = matrix4f;
+                RenderSystem.textureMatrix = matrix4f;
                 RenderSystem.vertexSorting = vertexSorting;
             });
         } else {
@@ -762,7 +_,7 @@
     public static void setTextureMatrix(Matrix4f textureMatrix) {
         Matrix4f matrix4f = new Matrix4f(textureMatrix);
         if (!isOnRenderThread()) {
-            recordRenderCall(() -> textureMatrix = matrix4f);
+            recordRenderCall(() -> RenderSystem.textureMatrix = matrix4f);
         } else {
             RenderSystem.textureMatrix = matrix4f;
         }
