--- a/net/minecraft/client/renderer/blockentity/BeaconRenderer.java
+++ b/net/minecraft/client/renderer/blockentity/BeaconRenderer.java
@@ -105,10 +_,13 @@
         float f10 = 1.0F;
         float f11 = -1.0F + fraction;
         float f12 = height * textureScale * (0.5F / beamRadius) + f11;
+        float finalF = f5;
+        float finalF1 = f12;
+        float finalF2 = f11;
         nodeCollector.submitCustomGeometry(
             poseStack,
             RenderType.beaconBeam(beamLocation, false),
-            (pose, consumer) -> renderPart(pose, consumer, color, yOffset, i, 0.0F, beamRadius, beamRadius, 0.0F, f5, 0.0F, 0.0F, f8, 0.0F, 1.0F, f12, f11)
+            (pose, consumer) -> renderPart(pose, consumer, color, yOffset, i, 0.0F, beamRadius, beamRadius, 0.0F, finalF, 0.0F, 0.0F, f8, 0.0F, 1.0F, finalF1, finalF2)
         );
         poseStack.popPose();
         f1 = -glowRadius;
@@ -119,11 +_,14 @@
         f10 = 1.0F;
         f11 = -1.0F + fraction;
         f12 = height * textureScale + f11;
+        float finalF3 = f1;
+        float finalF4 = f4;
+        float finalF5 = f5;
         nodeCollector.submitCustomGeometry(
             poseStack,
             RenderType.beaconBeam(beamLocation, true),
             (pose, consumer) -> renderPart(
-                pose, consumer, ARGB.color(32, color), yOffset, i, f1, f2, glowRadius, f4, f5, glowRadius, glowRadius, glowRadius, 0.0F, 1.0F, f12, f11
+                pose, consumer, ARGB.color(32, color), yOffset, i, finalF3, f2, glowRadius, finalF4, finalF5, glowRadius, glowRadius, glowRadius, 0.0F, 1.0F, finalF1, finalF2
             )
         );
         poseStack.popPose();
