--- a/net/minecraft/client/renderer/item/ConditionalItemModel.java
+++ b/net/minecraft/client/renderer/item/ConditionalItemModel.java
@@ -1,5 +_,6 @@
 package net.minecraft.client.renderer.item;
 
+import com.mojang.serialization.Codec;
 import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import javax.annotation.Nullable;
@@ -72,7 +_,7 @@
         }
 
         private static <T extends ConditionalItemModelProperty> T swapContext(T property, RegistryContextSwapper contextSwapper, ClientLevel level) {
-            return (T)contextSwapper.swapTo(property.type().codec(), property, level.registryAccess()).result().orElse(property);
+            return (T)contextSwapper.swapTo((Codec<T>) property.type().codec(), property, level.registryAccess()).result().orElse(property);
         }
 
         @Override
